File: src/arrays_and_strings/java/StringRotation.java
package arrays_and_strings.java;

/**
 *
 */
public class StringRotation {
  public static void main(String[] args) {
    String a = "waterbottle";
    String b = "erbottlewat";

    System.out.println(isRotation(a, b) ? "Yes" : "No");
  }

  private static boolean isRotation(String first, String second) {
    return first != null
        && second != null
        && first.length() > 0
        && first.length() == second.length()
        && isSubstring(first + first, second);
  }

  private static boolean isSubstring(String string, String subString) {
    return string != null && subString != null && string.contains(subString);
  }
}

File: src/arrays_and_strings/kotlin/StringRotation.kt
package arrays_and_strings.kotlin

/**
 *
 */
fun main(args: Array<String>) {
  val a = "waterbottle"
  val b = "erbottlewat"

  print(if (isRotation(a, b)) "Yes" else "No")
}

fun isRotation(first: String, second: String): Boolean {
  return first.isNotEmpty() && first.length == second.length && isSubString(first + first, second)
}

fun isSubString(first: String, second: String): Boolean {
  return first.isNotEmpty() && first.indexOf(second) in first.indices
}